std/concurrency.d

```diff
@@ -2473,6 +2474,23 @@ private
 
 version( unittest )
 {
+    void changeScheduler(Scheduler s)
+    {
+        import core.memory : GC;
+        import core.thread : Thread;
+
+        bool dosleep;
+        while (Thread.getAll().length != 1)
+        {
+            if (dosleep)
+                Thread.sleep(dur!("msecs")( 10 ));
+            else
+                dosleep = true;
+            GC.collect;
```

Comment:
Thanks for the explanation, I actually meant a comment in the source. I suggest something like `// wait for all other threads to terminate, using GC.collect to trigger finalizers which may terminate threads (e.g. OwnerTerminated or LinkTerminated)` at the top of the loop.